File: TTObjectClasses/Tıbbi Süreç/Kayıt - Kabul Modülü/Requirements/PatientInfoRequirements.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TTUtils.RequirementManager;
using System.ComponentModel;

namespace TTObjectClasses.Tıbbi_Süreç.Kayıt___Kabul_Modülü.Requirements
{
    public class PatientInfoRequirements : IRequirement
    {
        Patient patient = null;
        public PatientInfoRequirements(Patient patientParam)
        {
            patient = patientParam;

        }

        public RequirementResultCode ExecuteRequirement()
        {
            RequirementResultCode requirementResultCode = new RequirementResultCode();
            requirementResultCode.resultCode = 0;
            requirementResultCode.resultMessage = "Success";

            if (patient == null)
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M26893", "Sistemde hasta tanımlı değildir.");
                return requirementResultCode;
            }

            if (Convert.ToBoolean(TTObjectClasses.SystemParameter.GetParameterValue("CHECKMERNISNUMBER", "TRUE")) == true)
            {
                if (!Common.CheckMernisNumber(Convert.ToInt64(patient.UniqueRefNo)))
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25693", "Geçersiz TC Kimlik Numarası.");
                    return requirementResultCode;
                }
            }
                if (patient.Name == null)
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25107", "Ad alanı boş bırakılamaz.");
                return requirementResultCode;
            }

            if (patient.Surname == null)
            {
                requirementResultCode.resultCode = 1;
                requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M26943", "Soyad alanı boş bırakılamaz.");
                return requirementResultCode;
            }
                        
            if (patient.UnIdentified == false)
            {
                if (patient.MotherName == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25185", "Anne adı alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.FatherName == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25221", "Baba adı alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.BirthDate == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25515", "Doğum tarihi alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if(patient.BirthDate>Common.RecTime())
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25516", "Doğum tarihi alanı bugünün tarihinden büyük olamaz.");
                    return requirementResultCode;
                }
                
                if (patient.Sex == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25366", "Cinsiyet alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.Nationality == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M27147", "Uyruk alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.BirthPlace == null && patient.BirthPlace == "" && patient.Nationality.Kodu == "TR")
                {
                    requirementResultCode.resultCode = 7;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25518", "Doğum yeri alanı boş bırakılamaz.");
                    return requirementResultCode;
                }
                       
                if (patient.OtherBirthPlace == null &&  patient.OtherBirthPlace == "" && patient.Nationality.Kodu != "TR" )
                {
                    requirementResultCode.resultCode = 7;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25518", "Doğum yeri alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.Death == true && patient.ExDate == null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M26663", "Ölüm tarihi alanı boş bırakılamaz.");
                    return requirementResultCode;
                }

                if (patient.Death == true && patient.ExDate > Common.RecTime())
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M26664", "Ölüm tarihi alanı bugünün tarihinden büyük olamaz.");
                    return requirementResultCode;
                }
                if (patient.Death == true && patient.BirthDate > patient.ExDate)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25517", "Doğum tarihi alanı ölüm tarihinden büyük olamaz.");
                    return requirementResultCode;
                }

                //Hasta sistemde kayıtlı ise hata dönmeli
                if (patient.UniqueRefNo != null)
                {
                    BindingList<Patient> patientList = Patient.GetPatientByUniqueRefNoAndObjID(patient.ObjectContext, (long)(patient.UniqueRefNo.Value),patient.ObjectID);
                    if (patientList.Count > 0)
                    {
                       /* if (patientList.Count == 2)
                        {
                            Patient p = (Patient)patientList[1];
                            if (p.UniqueRefNo.Value == this.patient.UniqueRefNo.Value && p.Name != this.patient.Name && p.Surname != this.patient.Surname && p.MotherName != this.patient.MotherName && p.FatherName != this.patient.MotherName)
                            {
                                requirementResultCode.resultCode = 1;
                                requirementResultCode.resultMessage = "Bu TC Kimlik numarasına ait hasta dosyası bulunmaktadır.Hasta dosya no : " + p.ID;
                                return requirementResultCode;
                            }
                        }
                        else */
                        if (patientList.Count == 1)
                        {
                            Patient p = (Patient)patientList[0];
                            if (p.UniqueRefNo.Value == patient.UniqueRefNo.Value)
                            {
                                requirementResultCode.resultCode = 1;
                                requirementResultCode.resultMessage = "Bu TC Kimlik numarasına ait hasta dosyası bulunmaktadır.Hasta dosya no : " + p.ID;
                                return requirementResultCode;
                            }
                        }
                        else
                        {
                            requirementResultCode.resultCode = 1;
                            requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M25341", "Bu TC Kimlik numarasına ait birden fazla hasta dosyası bulunmaktadır.Dosya eşleştirme yapınız.");
                            return requirementResultCode;
                        }
                    }
                }

                //Gizli Hasta kontrolü
                if (patient.Privacy != null)
                {
                    if (patient.Privacy == true)
                    {
                        if (patient.PrivacyName == null)
                        {
                            requirementResultCode.resultCode = 1;
                            requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M00000", "'Rumuz ad' bilgisi giriniz.");
                            return requirementResultCode;
                        }
                        if (patient.PrivacySurname == null)
                        {
                            requirementResultCode.resultCode = 1;
                            requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M00000", "'Rumuz soyad' bilgisi giriniz.");
                            return requirementResultCode;
                        }
                    }
                }

               
            }
            else if(patient.UnIdentified == true)
            {
                if (patient.UniqueRefNo != null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M00000", "Kimlik Numarası alanı dolu olan hastalarda 'Kimliği belirsiz' işaretlenemez.");
                    return requirementResultCode;
                }
                if (!string.IsNullOrEmpty(patient.PassportNo))
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M00000","Pasaport No alanı dolu olan hastalarda 'Kimliği belirsiz' işaretlenemez.");
                    return requirementResultCode;
                }
                if (patient.YUPASSNO != null)
                {
                    requirementResultCode.resultCode = 1;
                    requirementResultCode.resultMessage = TTUtils.CultureService.GetText("M00000", "YUPASS No alanı dolu olan hastalarda 'Kimliği belirsiz' işaretlenemez.");
                    return requirementResultCode;
                }
            }
            return requirementResultCode;
        }
    }
  
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Kayit_Kabul_Modulu/Requirements/PatientInfoRequirements.ts
//$7F76CE1D
import { RequirementResultCode } from "Fw/Requirements/RequirementResultCode";
import { IRequirements } from "Fw/Requirements/IRequirements";
import { Patient } from 'NebulaClient/Model/AtlasClientModel';

export class PatientInfoRequirements implements IRequirements {


    public patient: Patient;
    public PatientInfoUIElements: Map<string, any>;
    public requiredColor: string = "#f3d7d7";
    public normalColor: string = "white";
    public name: String;
    public surname: String;
    constructor(patientParam: Patient, nameParam: String, surnameParam: String, controls: Map<string, any>) {
        this.patient = patientParam;
        this.name = nameParam;
        this.surname =  surnameParam;
        this.PatientInfoUIElements = controls;
    }

    ExecuteRequirementsWithApproval(): Promise<RequirementResultCode> {
        throw new Error("Method not implemented.");
    }
    public ExecuteRequirements(): RequirementResultCode {

        let requirementResultCode: RequirementResultCode = new RequirementResultCode();
        requirementResultCode.resultCode = 0;
        requirementResultCode.resultMessage = "Success";
        let UIComponent = null;

        if (this.patient == null) {
            requirementResultCode.resultCode = 1;
            requirementResultCode.resultMessage = i18n("M21923", "Sistemde hasta tanımlı değildir.");
            return requirementResultCode;
        }

        if (this.patient.UnIdentified == false) {

            if (this.patient.MotherName == null || this.patient.MotherName == "") {
                requirementResultCode.resultCode = 4;
                requirementResultCode.resultMessage = i18n("M11038", "Anne adı alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("MotherName");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if (this.patient.FatherName == null || this.patient.FatherName == "") {
                requirementResultCode.resultCode = 5;
                requirementResultCode.resultMessage = i18n("M11392", "Baba adı alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("FatherName");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if (this.patient.BirthDate == null) {
                requirementResultCode.resultCode = 6;
                requirementResultCode.resultMessage = i18n("M13134", "Doğum tarihi alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("BirthDate");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }


            if (this.patient.BirthPlace == null && this.patient.Nationality != null && this.patient.Nationality.Kodu  == "TR" ) {
                requirementResultCode.resultCode = 7;
                requirementResultCode.resultMessage = i18n("M13143", "Doğum yeri alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("CityOfBirth");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if (this.patient.OtherBirthPlace == null && ( this.patient.Nationality == null || this.patient.Nationality.Kodu != "TR")) {
                requirementResultCode.resultCode = 7;
                requirementResultCode.resultMessage = i18n("M13143", "Doğum yeri alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("OtherBirthPlace");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if (this.patient.Sex == null || this.patient.Sex.ADI == null) {
                requirementResultCode.resultCode = 8;
                requirementResultCode.resultMessage = i18n("M12268", "Cinsiyet alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("Sex");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if (this.patient.Nationality == null) {
                requirementResultCode.resultCode = 9;
                requirementResultCode.resultMessage = i18n("M23816", "Uyruk alanı boş bırakılamaz.");
                UIComponent = this.PatientInfoUIElements.get("Nationality");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }

            if(this.patient.UniqueRefNo != null && this.patient.UniqueRefNo.toString().startsWith("99") && this.patient.Nationality.Kodu == "TR"){
                requirementResultCode.resultCode = 10;
                requirementResultCode.resultMessage = ("Kimlik numarası 99 ile başlayan hastalar için uyruk 'Türkiye Cumhuriyeti' Seçilemez. ");
                UIComponent = this.PatientInfoUIElements.get("Nationality");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }
        }
        else if (this.patient.UnIdentified == true)
        {
            if (this.patient.ImportantPatientInfo == null) {
                requirementResultCode.resultCode = 11;
                requirementResultCode.resultMessage = i18n("M23816", "Lütfen Hasta Detay Bilgileri Tabındaki 'Arşiv Açıklaması' alanını doldurunuz.");
                UIComponent = this.PatientInfoUIElements.get("ImportantPatientInfo");
                if (UIComponent != null)
                    UIComponent.BackColor = this.requiredColor;
            }
        }

        if (this.name == null) {
            requirementResultCode.resultCode = 2;
            requirementResultCode.resultMessage = i18n("M10497", "Ad alanı boş bırakılamaz.");
            UIComponent = this.PatientInfoUIElements.get("Name");
            if (UIComponent != null)
                UIComponent.BackColor = this.requiredColor;

        }

        if (this.surname == null) {
            requirementResultCode.resultCode = 3;
            requirementResultCode.resultMessage = i18n("M22203", "Soyad alanı boş bırakılamaz.");

            UIComponent = this.PatientInfoUIElements.get("Surname");
            if (UIComponent != null)
                UIComponent.BackColor = this.requiredColor;
        }

        return requirementResultCode;

    }

}
